metals/src/main/scala/scala/meta/internal/builds/ScalaCliBuildTool.scala

```diff
@@ -11,9 +11,14 @@ import scala.meta.internal.metals.UserConfiguration
 import scala.meta.internal.metals.scalacli.ScalaCli
 import scala.meta.io.AbsolutePath
 
+import org.eclipse.lsp4j.MessageParams
+import org.eclipse.lsp4j.MessageType
+import org.eclipse.lsp4j.services.LanguageClient
+
 class ScalaCliBuildTool(
     val workspaceVersion: Option[String],
     userConfig: () => UserConfiguration,
+    languageClient: LanguageClient,
```

Comment:
I don't really like having `languageClient` as a parameter here, maybe we can do `showMessage` in `maybeAmendScalaCliBspConfig`?

Reply:
I hear you. It is quite awkward. I'm just not sure how to do this better.

We call `generateBspConfig` with `systemProcess  = BspConfigGenerator.runUnconditionally`, which processes the args. In `BspConfigGenerator.runUnconditionally` we show to the user `slowTask` w/ msg `build tool bspCofig` and there is also a status bar shown. However, for `scala-cli` we fallback to creating the bsp config "by hand" and none of those prompts will be shown to the user, which feels just a little inconsistent.

I could try and change the consumer type (`systemProcess`) allow fallback...